addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoHandlerFactory.java

```diff
@@ -35,9 +66,59 @@ protected ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
         if (thingTypeUID.equals(THING_TYPE_ONKYOAV) || thingTypeUID.equals(THING_TYPE_ONKYO_UNSUPPORTED)) {
-            return new OnkyoHandler(thing);
+            String callbackUrl = createCallbackUrl();
```

Comment:
If the callbackUrl is null, you probably don't want to go on and register the handler as an AudioSink.